src/main/kotlin/fi/metatavu/pakkasmarja/services/erp/impl/translate/BusinessPartnerTranslator.kt

```diff
@@ -0,0 +1,111 @@
+package fi.metatavu.pakkasmarja.services.erp.impl.translate
+
+import com.fasterxml.jackson.databind.JsonNode
+import fi.metatavu.pakkasmarja.services.erp.api.model.SapAddress
+import fi.metatavu.pakkasmarja.services.erp.api.model.SapAddressType
+import fi.metatavu.pakkasmarja.services.erp.api.model.SapBankAccount
+import fi.metatavu.pakkasmarja.services.erp.api.model.SapBusinessPartner
+import java.time.*
+import javax.enterprise.context.ApplicationScoped
+
+/**
+ * Translator class for SAP business partners
+ */
+@ApplicationScoped
+class BusinessPartnerTranslator: AbstractTranslator<JsonNode, SapBusinessPartner>() {
+    /**
+     * Translates a business partner from SAP into a format expected by spec
+     *
+     * @param entity business partner to be translated
+     *
+     * @return translated business partner
+     */
+    override fun translate(entity: JsonNode): SapBusinessPartner {
+        return SapBusinessPartner(
```

Comment:
This code assumes quite a lot, is it really so that all of there fields exist on every single response? Some sort of error handling for these would be good.

Reply:
Added